hack/generator/pkg/astmodel/armconversion/shared.go

```diff
@@ -129,7 +129,12 @@ type complexPropertyConversionParameters struct {
 	destinationType   astmodel.Type
 	nameHint          string
 	conversionContext []astmodel.Type
-	assignmentHandler func(result ast.Expr, destination ast.Expr) ast.Stmt
+	assignmentHandler func(destination, source ast.Expr) ast.Stmt
+
+	// sameTypes indicates that the source and destination types are
```

Comment:
Rather than "base-level" say something about "type"? Basically this is saying: "If the types are the same we don't need to worry about renaming any of the `TypeName`'s"?

Reply:
Good call, I've reworded it to say "so no conversion between Arm and non-Arm types is required (although structure copying is)".